### ISSUE TITLE
Parallelize bc compilation

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#4664]
### Steps to Reproduce

1. Build for watchOS/Release/Device.
2. Wait & wait & wait a long time until it completes.
3. Look at the build log, and realize that we pass all .bc files to clang when linking the final executable, and this will make clang compile each .bc file serially, which takes a long time.

This was discovered by Zoltan (https://github.com/mono/mono/issues/9867#issuecomment-412188621).

It would be much better if we compiled all the .bc files to .o files in parallel, and then finally just linked the .o files together.

Incomplete patch: https://github.com/rolfbjarne/xamarin-macios/commit/74ab6d9b2350f025c5fa5280fac7bbf8f97dc40f

This patch makes mtouch compile dontlink/watchOS/Release/Device in 50s instead of 100s.

Missing bits:

* Tests (in particular regarding custom clang flags).

**Comment by rolfbjarne** [comment]
This is no longer relevant, since Apple is removing bitcode.
